File: civitasjava/src/Civitas/CasillaSorpresa.java
package civitas;

import java.util.ArrayList;

public class CasillaSorpresa extends Casilla{
    private Sorpresa sorpresa;
    private MazoSorpresas mazo;
    
    CasillaSorpresa(MazoSorpresas m, String n){
        super(n);
        this.mazo = m;
    }
    
    @Override
    public void recibeJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual, todos)){
            sorpresa = mazo.siguiente();
            informe(actual, todos);
            sorpresa.aplicarAJugador(actual, todos);
        }
    }
    
    @Override
    public String toString(){
        return "CasillaSorpresa{ Sorpresa: " +sorpresa + ", mazo: " + mazo +"}";
    }
}

File: RubyCivitas/lib/casilla_sorpresa.rb
require_relative 'casilla'
require_relative 'sorpresa'

module Civitas
  class Casilla_sorpresa < Casilla
    
    def initialize(mazo, nombre)
      
      super(nombre)
      @mazo = mazo
      
    end
    
    #Override
    def recibe_jugador(actual, todos)

      if jugador_correcto(actual, todos)

        @sorpresa = @mazo.siguiente
        
        informe(actual, todos)
        
        #puts @sorpresa
        @sorpresa.aplicar_al_jugador(actual, todos)

      end
    end
    
    #Override
    def to_string
      
      "[Casilla] -> #{@nombre}\n[Tipo] -> Sorpresa\n[Sorpresa] -> #{@sorpresa.to_string}"
      
    end
   
  end
end
